### PR TITLE
Add typedef ranges

### PR DESCRIPTION
**Pull request by RRozak**
Continuation of https://github.com/antmicro/verilator/pull/270/

### CODE REVIEW

**Comment by RRozak** [comment]
Blocked by https://github.com/chipsalliance/Surelog/issues/1412

**Comment by RRozak** [comment]
Typedef statements with ranges require https://github.com/chipsalliance/Surelog/pull/1424

File: src/UhdmAst.cpp [hunk 1]
```diff
@@ -487,6 +492,25 @@ AstNodeDType* getDType(FileLine* fl, vpiHandle obj_h, UhdmShared& shared) {
...
+    case vpiArrayTypespec: {
+        AstRange* rangeNodep = nullptr;
+        std::stack<AstRange*> range_stack;
+        visit_one_to_many({vpiRange}, obj_h, shared, [&](AstNode* node) {
+            rangeNodep = reinterpret_cast<AstRange*>(node);
+            range_stack.push(rangeNodep);
+        });
+
+        auto elem_typespec_h = vpi_handle(vpiElemTypespec, obj_h);
+        AstNodeDType* dtypep = getDType(fl, elem_typespec_h, shared);
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: We already have `AstNodeDType* dtype` here - which, as a matter of fact, should have been called `dtypep`. Can we just use that (and maybe rename it)?

File: src/UhdmAst.cpp [hunk 2]
```diff
@@ -487,6 +492,25 @@ AstNodeDType* getDType(FileLine* fl, vpiHandle obj_h, UhdmShared& shared) {
         if (dtype == nullptr) { dtype = basic; }
         break;
     }
+    case vpiArrayTypespec: {
+        AstRange* rangeNodep = nullptr;
+        std::stack<AstRange*> range_stack;
+        visit_one_to_many({vpiRange}, obj_h, shared, [&](AstNode* node) {
+            rangeNodep = reinterpret_cast<AstRange*>(node);
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: `node`->`nodep`

File: src/UhdmAst.cpp [hunk 3]
```diff
@@ -645,38 +668,34 @@ AstNodeDType* getDType(FileLine* fl, vpiHandle obj_h, UhdmShared& shared) {
...
+            if (typespec_h) {
+                dtype = getDType(fl, typespec_h, shared);
             } else {
                 auto element_type = vpi_get(vpiType, member_h);
                 if (element_type) {
                     AstBasicDTypeKwd keyword = get_kwd_for_type(element_type);
-                    elementDtypep = new AstBasicDType(fl, keyword);
+                    dtype = new AstBasicDType(fl, keyword);
+                } else {
+                    v3error("Missing typespec for unpacked_array_var");
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: Match the VPI/UHDM name here: `array_var`.

**Review by rkapuscik — CHANGES_REQUESTED** [review]

File: src/UhdmAst.cpp (hunk 2 above)
Comment (RRozak) [reply src/UhdmAst.cpp]: Fixed

File: src/UhdmAst.cpp (hunk 3 above)
Comment (RRozak) [reply src/UhdmAst.cpp]: Fixed

File: src/UhdmAst.cpp (hunk 1 above)
Comment (RRozak) [reply src/UhdmAst.cpp]: Done, I renamed it in the whole function

**Review by rkapuscik — APPROVED** [review]

**Pull request merged**